### PR TITLE
Promote cloud-init schema (SC-953)

### PR DESCRIPTION
**Pull request by holmanb**
```
Promote schema command from devel to top level subcommand
```

### CODE REVIEW

File: cloudinit/cmd/devel/parser.py
Comment (TheRealFalcon) [inline cloudinit/cmd/devel/parser.py:33]: We have a warning with our devel commands that they can and will change (and likely be promoted top-level), so I'm ok removing this entirely rather than deprecating it.

**Review by TheRealFalcon — CHANGES_REQUESTED** [review]
Some really nice changes here. Normally I like small updates along the way, but it looks like there's more non-schema changes than there are schema ones. Do you mind moving the misc updates to another PR?

Additionally, there's some other places that need to be updated.
* If you search for "devel schema" in faq.rst, tutorial.rst, and tests/integration_tests/modules/test_cli.py, they all need updates.
* Search '"devel", "schema"' in tests/unittests/test_cli.py
* cli.rst has schema documentation under the devel heading.
* `bash_completion/cloud-init` has schema is under devel

**Comment by holmanb** [comment]
I had to rebase for dropping commits, hence the force push. Since I already had to force push, I squashed the recommended "drop devel schema" change into the promotion commit since they overlapped.

**Review by TheRealFalcon — APPROVED** [review]
LGTM, thanks!

I trust you can fix the flake8 and spell check issue without another review.

**Comment by holmanb** [comment]
After rerunning CI passes, merging.

**Pull request merged**
